cmd/zed/agents/zfs_mod.c

```diff
@@ -682,12 +706,16 @@ zfs_deliver_add(nvlist_t *nvl, boolean_t is_lofi)
 	 * Iterate over all vdevs looking for a match in the following order:
 	 * 1. ZPOOL_CONFIG_DEVID (identifies the unique disk)
 	 * 2. ZPOOL_CONFIG_PHYS_PATH (identifies disk physical location).
-	 *
-	 * For disks, we only want to pay attention to vdevs marked as whole
-	 * disks or are a multipath device.
+	 * 3. ZPOOL_CONFIG_GUID (identifies unique vdev).
 	 */
-	if (!devid_iter(devid, zfs_process_add, is_slice) && devpath != NULL)
-		(void) devphys_iter(devpath, devid, zfs_process_add, is_slice);
+	if (devid_iter(devid, zfs_process_add, is_slice))
+		return (0);
+	if (devpath != NULL && devphys_iter(devpath, devid, zfs_process_add,
+	    is_slice))
+		return (0);
+	if (vdev_guid != 0)
```

Comment:
It seems like we should be checking that `pool_guid != 0` here as well.

Reply:
If pool_guid is 0, zed looks at every pool. I'm not sure that's a bad thing so I let it happen, but if it would be better not to I can make this change.

Reply:
I'm not sure there's ever a situation where we'd expect to get a vdev_guid but not a pool_guid.  But since passing a zero here means check every pool (which isn't harmful) this seems fine.  Even though I'd expect it never to happen.